server/routes/router.js

```diff
@@ -1,72 +1,54 @@
-var express = require('express');
-var router = express.Router();
-var axios = require('axios');
-var users = require("../models/userAuth");
-var mongoose = require('mongoose');
-
+const express = require('express');
+const router = express.Router();
+const axios = require('axios');
+const users = require('../models/userAuth');
+const mongoose = require('mongoose');
 
 //index router
-router.get('/' , (req,res) => {
+router.get('/', (req, res) => {
     console.log("Recived home request");
     res.send("Hello world!");
-
-    
-})
+});
 
 //demo route
 router.get('/home', (req, res)=>{
     console.log("home called");
     res.send("hello");
-} )
-
+});
 
 //route that handles register requests
-router.post("/register" , (req , res) => {
+router.post('/register', (req, res) => {
 
     //create a newUser document (ref: userAuth.js)
+    const { name, username, email, password, ph } = req.body;
     const newUser = new users({
-        name : req.body.name,
-        username : req.body.username,
-        email : req.body.email,
-        password : req.body.password,
-        ph : req.body.ph,
-    })
+        name, username, email, password, ph
+    });
```

Comment:
@mashuDuek Can you provide any relevant materials showing that the schema constructor supports this method of initialization i.e. without key-value pairs?

Because as far as the mongoose documentation and other relevant online resources go, they all seem to adhere to this key-value scheme. 

Also, chances are schema constructors don't support self-resolution i.e. they need key-value pairs . 

Feel free to drop any relevant resources/code

Reply:
Having a rough time finding an example that uses ES6 object creation to instantiate a mongoose model. 

What is passed is as an argument to `new users` is the same thing it was before, it shouldn't have changed functionality. [MDN Object Initializer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer)

![Screen Shot](https://user-images.githubusercontent.com/26696628/64378987-f115f600-cffb-11e9-8400-f2f2826f44bf.png)

Reply:
Was this not good enough documentation on this?